### DISCUSSION TITLE
Automatic greetings when a user accesses a private group via link

### DISCUSSION THREAD

**Discussion by Pro-vir-2223** [telegraf/telegraf#1765] [Q&A]
Good afternoon,

I have been using the following code, so that when a user accesses through an invitation link, automatically the bot greets him by his name.

bot.on('new_chat_members', (ctx) => {
  ctx.reply(`Hello, ${ctx.from.first_name}. 
  `);
})

The problem is that this only works if the group is in public, because if the group is in private, it does not work when a user accesses the group,

![image](https://user-images.githubusercontent.com/121511611/209701447-cc9c34ea-3f52-40f0-91ba-2afe3eaa3e6b.png)

On the other hand, if I activate approve new members, when a new member is approved, the notification appears.

In short, we need that in a private group, the script can work without having to set it to "Aprove new members".

![image](https://user-images.githubusercontent.com/121511611/209701642-2b0b0fe3-9e05-4768-a3d7-66e3e778a98a.png)

Thank you very much -^^-
